### ISSUE TITLE
Intersection type unexpectedly narrows to `never`

### ISSUE THREAD

**Issue by NoelAbrahams** [microsoft/TypeScript#38263]
**TypeScript Version:**  3.9 RC

**Search Terms:**
intersection never 3.9 

**Code**

```ts
enum Foo {

        original = 1,
        modified = 2,
   }

type Original = {

    type: Foo.original;
    title: string;
};

type Modified = {

    type: Foo.modified;
    modification: string;
};

type Bar = Original & Modified;
var x = {} as Bar;

x.title; // Error Property 'title' does not exist on type 'never'

```

**Expected behavior:**
No error

**Actual behavior:**
An error is reported.

**Playground Link:**

[Playground link](https://www.typescriptlang.org/play/?ts=4.0.0-dev.20200429&ssl=1&ssc=1&pln=24&pc=1#code/KYOwrgtgBAYg9nKBvAsAKHVL2twE4CWA5gSAIYA2UAvFAIwA0mO2EcAJgQGYHDs1QATEzTYAvunQAXAJ4AHYFADyhEuSq1UGUdlkKAXLAQA6fMVKUA3MyxSCUisEMBnKYRBFraMV+nzFALIc3Lz8mpI6tv6G8HDGbJw8fF44CSEAxmR2cCAubqSe6D4ReooAQmR4Airm6lAAZFBBiaFeAG6VUAAeAkhiUGTOUBV4vmhdxnYOwJZQAPRzUACieHj4UAAKawp4slAA5FOO+1DscMBDIHBSUMBdBK5QOVClByDAbcB4+xFAA)

**Related Issues:**
https://github.com/microsoft/TypeScript/issues/38162

**Comment by NoelAbrahams** [comment]
I see that this is an intentional breaking change.
